### DISCUSSION TITLE
Updating from 5.3.31 to 5.4.22 fails

### DISCUSSION THREAD

**Discussion by TopperDEL** [unoplatform/uno#18321] [Q&A] [answered]
According to [this](https://aka.platform.uno/upgrade-uno-packages) link I should only change the Sdk-value in global.json to the new version. But on Nuget the Uno.Sdk-max-Version is 5.4.5, whereas I expected it to be 5.4.22.

If I switch to the latest 5.4.5 Uno.Sdk I get the following error:
![image](https://github.com/user-attachments/assets/a26b0be3-492d-4944-a5f0-20725707087e)

And this is my csproj:
```
<Project Sdk="Uno.Sdk">
  <PropertyGroup>
    <TargetFrameworks>
      <!--net8.0-maccatalyst;-->
      net8.0-windows10.0.19041;
      net8.0-desktop;
      <!--net8.0-browserwasm;-->
    </TargetFrameworks>

    <OutputType>Exe</OutputType>
    <UnoSingleProject>true</UnoSingleProject>
    <ManagePackageVersionsCentrally>true</ManagePackageVersionsCentrally>

    <!-- Display name -->
    <ApplicationTitle>MyApp</ApplicationTitle>
    <!-- App Identifier -->
    <ApplicationId>com.myapp</ApplicationId>
    <!-- Versions -->
    <ApplicationDisplayVersion>1.0</ApplicationDisplayVersion>
    <ApplicationVersion>1</ApplicationVersion>
    <!--
      If you encounter this error message:

        error NETSDK1148: A referenced assembly was compiled using a newer version of Microsoft.Windows.SDK.NET.dll.
        Please update to a newer .NET SDK in order to reference this assembly.

      This means that the two packages below must be aligned with the "build" version number of
      the "Microsoft.Windows.SDK.BuildTools" package above, and the "revision" version number
      must be the highest found in https://www.nuget.org/packages/Microsoft.Windows.SDK.NET.Ref.
    -->
    <!-- <WindowsSdkPackageVersion>10.0.22621.28</WindowsSdkPackageVersion> -->

    <!--
      UnoFeatures let's you quickly add and manage implicit package references based on the features you want to use.
      https://aka.platform.uno/singleproject-features
    -->
    <UnoFeatures>
      Material;
      Hosting;
      Toolkit;
      Logging;
      Mvvm;
      ExtensionsCore;
    </UnoFeatures>
  </PropertyGroup>

[..]
```

**Comment by jeromelaban** [comment]
Thanks for the report. This is a WinAppSDK temporary issue, make sure to follow https://platform.uno/docs/articles/migrating-from-previous-releases.html#winappsdk-16-considerations for upgrading.

**Reply by TopperDEL** [reply]
Doing what suggested brings this error:
![image](https://github.com/user-attachments/assets/c59b36ea-c36d-4a79-94f5-7aca2650758a)

**Reply by jeromelaban** [reply]
WinAppSDK includes WebView2 in 1.6, and you probably have it included in your solution. Removing it is likely to fix the issue.

**Reply by TopperDEL** [reply]
It is marked as "transient", so I did not include it explicitly.

**Reply by jeromelaban** [reply]
Thanks. We'll have to find which is importing it, and you may be able to find the origin using a binlog.

**Reply by TopperDEL** [reply]
I've created a binlog, but I'm unable to see where the reference comes from. Maybe you can see it @jeromelaban ?
[msbuild.zip](https://github.com/user-attachments/files/17201555/msbuild.zip)

**Reply by jeromelaban** [reply]
This may be because there's another project below that is referencing a different version of WinAppSDK, or maybe just a bug of WinAppSDK that duplicates the use of that particular binary. It's likely that creating winui-only repro and posting to the WinAppSDK repository will be needed.

You could also create an msbuild target that alters the build to remove the duplicate file.

**Reply by MartinZikmund** [reply, marked as answer]
@TopperDEL please try doing full `git clean -xfd` and see if it helps - I got this error as well after updating, but it seems that it was somewhere in temp files or bin/obj and after clean it seems to work fine again

**Reply by TopperDEL** [reply]
@MartinZikmund That did the trick!
I also had to add this PropertyGroup as mentioned in the docs, but then I could update to 5.4:

```
<PropertyGroup>
    <!-- Remove when the .NET SDK 8.0.403 or later has been released -->
    <WindowsSdkPackageVersion>10.0.19041.38</WindowsSdkPackageVersion>
</PropertyGroup>
```

Thank you both!
